### DISCUSSION TITLE
Lesson 11 opensea/pinata

### DISCUSSION THREAD

**Discussion by Callassie** [smartcontractkit/full-blockchain-solidity-course-py#1380] [General]
Have been through all the posts regarding the metadata/Opensea not showing. None of the fixes worked for me.  

I went ahead and pinned on Pinata as my IPFS daemon keeps going down, both metadata and json. Does that make a difference?
Assuming that Pinata is just a pinning service and utilizes the IPFS hash and we don't need to change anything. Here is the link to opensea: https://testnets.opensea.io/assets/0x0894ab9b876e22e9069498485a5ad4cc228f9ea4/2

I would like to move on to the proxy lesson, but I can't!

**Comment by cromewar** [comment]
Hello @Callassie 
Just a question, the problem is just showing up on opensea? or you cannot access to the image also with the URI link?

**Reply by Callassie** [reply]
I am able to see the image here:
https://ipfs.io/ipfs/QmYx6GsYAKnNzZ9A6NvEKV9nf1VaDzJrqDR23Y8YSkebLU?filename=shiba-inu.png

and here

https://ipfs.io/ipfs/QmSsYRx3LpDAb1GZQm7zZ1AuHZjfbPkD6J7s9r41xu1mf8?filename=pug.png

**Comment by Callassie** [comment]
I just ran create_collectible again. So, ran create_metadata and set_tokenuri again. ( edited: IPFS is up and running. but also pinned to pinata just in case.)
Got this:
Working on rinkeby
You have 4 tokenIds
Setting tokenURI of 3
Transaction sent: 0x45d0868616ad58753f6fae42aac0d5cedf1bc524191cc4d861d23eae2448d180
  Gas price: 1.000000073 gwei   Gas limit: 132237   Nonce: 143
  AdvancedCollectible.setTokenURI confirmed   Block: 10469037   Gas used: 120216 (90.91%)

  AdvancedCollectible.setTokenURI confirmed   Block: 10469037   Gas used: 120216 (90.91%)

You can view your NFT at https://testnets.opensea.io/assets/0x0894aB9b876e22E9069498485A5aD4cc228f9ea4/3
Please wait up to 20 minutes and hit the refresh metadata button.

I will wait the 20 minutes, refresh both the metadata and the page and opensea will not show.

**Reply by cromewar** [reply]
By experience I can't say that it could take a lot mote than 20 minutes on the testnet, seems everything it ok on your side... So let's hope is a opensea bug

**Reply by Callassie** [reply]
I was thinking the same thing, but if you look at the test account page, it seems like something is not matching. Look at how many NFTs!
https://testnets.opensea.io/0xcfa07c83306de7f0814ebc1bde251eccc68c0725

I wrote a loop in the upload_to_pinata for both the images and the metadata. I am not learned in python, so I researched a few ways to read a directory and such...  seems like it is not picking up the last file in the directory. Maybe, something in that loop is causing this problem? Here is the loop code:
![screenShot1](https://user-images.githubusercontent.com/102427356/162477848-4fea7927-b397-4013-bce7-b0112bc14389.png)

**Reply by cromewar** [reply]
I'll try to reproduce the error,does you code match the GitHub repo?

**Reply by Callassie** [reply]
everything except for the upload_to_pinata

**Reply by Callassie** [reply]
I did change the description in the metadata just as a check, but when I replaced my code with the github code, it changed it back. So, that may be why there are so many nfts, with different metadata? Just a thought and before you do a lot of testing.

**Reply by cromewar** [reply]
As that nft was deployed for so many accounts before... Yes most likely you are right.

**Comment by Callassie** [comment]
I will move on to the next lesson, but it has not been definitively proven it is Opensea. After the course, I will use a new graphic, create a new project with the code and see what happens.Sent via the Samsung Galaxy S20 FE 5G, an AT&T 5G smartphone
-------- Original message --------From: Vasiliy Gualoto ***@***.***> Date: 4/8/22  12:38 PM  (GMT-06:00) To: smartcontractkit/full-blockchain-solidity-course-py ***@***.***> Cc: Callassie ***@***.***>, Mention ***@***.***> Subject: Re: [smartcontractkit/full-blockchain-solidity-course-py] Lesson 11 opensea/pinata (Discussion #1380) 
As that nft was deployed for so many accounts before... Yes most likely you are right.

**Reply by cromewar** [reply]
I tried to reproduce the error, but the NFT is showing correctly for me, so I'm kinda lost here.

**Comment by Callassie** [comment]
I appreciate the information. I went ahead and did the proxy lesson. I will do a full test with a different png, so it will be a different hash and test that. If I can't get through it. I will redo that lesson.Appreciate your help!Sent via the Samsung Galaxy S20 FE 5G, an AT&T 5G smartphone
-------- Original message --------From: Vasiliy Gualoto ***@***.***> Date: 4/8/22  8:07 PM  (GMT-06:00) To: smartcontractkit/full-blockchain-solidity-course-py ***@***.***> Cc: Callassie ***@***.***>, Mention ***@***.***> Subject: Re: [smartcontractkit/full-blockchain-solidity-course-py] Lesson 11 opensea/pinata (Discussion #1380) 
I tried to reproduce the error, but the NFT is showing correctly for me, so I'm kinda lost here.

**Comment by cromewar** [comment]
Well using a different PNG made no difference and I also worked, so could you just try cloning the repo and run it? does that work?

**Comment by Callassie** [comment]
I will. Currently on the front end on lesson 13Sent via the Samsung Galaxy S20 FE 5G, an AT&T 5G smartphone
-------- Original message --------From: Vasiliy Gualoto ***@***.***> Date: 4/10/22  10:27 AM  (GMT-06:00) To: smartcontractkit/full-blockchain-solidity-course-py ***@***.***> Cc: Callassie ***@***.***>, Mention ***@***.***> Subject: Re: [smartcontractkit/full-blockchain-solidity-course-py] Lesson 11 opensea/pinata (Discussion #1380) 
Well using a different PNG made no difference and I also worked, so could you just try cloning the repo and run it? does that work?

**Comment by Callassie** [comment]
Thank you for that information as well. I am hopeful to finish lesson 13 quickly and try this. For the front-end on Lesson 13, seems like my windows version had a lot of deprecated and mssing modules. Npx and yarn were there, but when I checked by running yarn after changing directories, it needed to install a lot of packages. I don't know if you need to know that for your Spanish version. Sent via the Samsung Galaxy S20 FE 5G, an AT&T 5G smartphone
-------- Original message --------From: Vasiliy Gualoto ***@***.***> Date: 4/10/22  10:27 AM  (GMT-06:00) To: smartcontractkit/full-blockchain-solidity-course-py ***@***.***> Cc: Callassie ***@***.***>, Mention ***@***.***> Subject: Re: [smartcontractkit/full-blockchain-solidity-course-py] Lesson 11 opensea/pinata (Discussion #1380) 
Well using a different PNG made no difference and I also worked, so could you just try cloning the repo and run it? does that work?

**Comment by ptn77** [comment]
Hi,
I'm having the same issue. Although I changed the code to use my own image. I can't seem to get the image to show up in opensea and it's been longer than 20 minutes.
Here it shows the set_uri was successful:
https://rinkeby.etherscan.io/address/0xeb6df80433b81fd866e67ef5b13bb9a283cb46b9
on this contract:
https://rinkeby.etherscan.io/address/0xb64ca1f58824f14ba29245dd2291a156c5d8247f
You can see the uri for token 0 as : https://gateway.pinata.cloud/ipfs/QmQ4L8uAZNnBGTe1two4iHTzDYrZ1bA1adBCxmvCMUMa1m

tokenId (uint256)
0
 string

[ tokenURI(uint256) method Response ]
    string :  https://gateway.pinata.cloud/ipfs/QmQ4L8uAZNnBGTe1two4iHTzDYrZ1bA1adBCxmvCMUMa1m

and I'm able to get to the image, but on opensea, it's not showing up:
https://testnets.opensea.io/assets/0xb64cA1f58824f14Ba29245dd2291a156C5D8247F/0

Is this expected? How long has it been taking for the images to show up on the opensea testnet?

thanks
